### PR TITLE
Add EKS integ test

### PR DESCRIPTION
**Pull request by hdj630**

### CODE REVIEW

File: src/main/java/com/amazon/aocagent/tasks/EKSClean.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                deploymentName, "default", "true", null, 0, null, "Foreground", null);
+      }
+      log.info("Deployment {} doesn't exist", deploymentName);
+    } catch (com.google.gson.JsonSyntaxException e) {
+      // Known issue https://github.com/kubernetes-client/java/issues/86
+      log.info("ignore JsonSyntaxException");
+    }
+
+    // wait for pods to be deleted
+    while (true) {
```
Comment (mxiamxia) [inline src/main/java/com/amazon/aocagent/tasks/EKSClean.java:70]: may need to consider the deletion failed case. open try for a certain amount of time?

Comment (hdj630) [reply src/main/java/com/amazon/aocagent/tasks/EKSClean.java:70]: OK

File: README.md [hunk 2]
```diff
@@ -75,6 +75,15 @@ gradle run --args="integ-test -t=ECS_TEST --package-version={the version you wan
 gradle run --args="clean -t=ECSClean --package-version={the version you want to test}"
 ```
 
+### Run EKS Integ-test on Sidecar mode (EMF Metrics)
+```
+gradle run --args="integ-test -t=EKS_TEST --package-version={the version you want to test} -k eksClusterName=my-cluster-name -k awsAuthenticatorPath=/my/authenticator/path"
```
Comment (mxiamxia) [inline README.md:80]: can we say ```awsAuthenticatorPath=/my/authenticator/path``` is optional? I see you will download the default one from aws  if customer doesn't provide it.

**Review by mxiamxia — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (hdj630) [reply README.md:80]: yes, so optional option can be absent here, correct?

File: src/main/java/com/amazon/aocagent/helpers/EKSTestOptionsValidationHelper.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+@Log4j2
+public class EKSTestOptionsValidationHelper {
+  /**
+   * validate EKS test options.
+   *
+   * @param context test context
+   */
+  public static void checkEKSTestOptions(Context context) {
+    if (context.getEksClusterName() == null) {
+      throw new RuntimeException("EKS test without specifying cluster name");
```
Comment (wyTrivail) [inline src/main/java/com/amazon/aocagent/helpers/EKSTestOptionsValidationHelper.java:20]: can we use baseexception here?

File: src/main/java/com/amazon/aocagent/helpers/EKSTestOptionsValidationHelper.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                    + binaryName);
+        try {
+          FileUtils.copyURLToFile(new URL(binaryUrl), binaryFile);
+        } catch (IOException e) {
+          throw new RuntimeException("Download aws-iam-authenticator " + binaryUrl + " failed.");
+        }
+        binaryFile.setExecutable(true);
+        context.setIamAuthenticatorPath(binaryFile.getPath());
+      } else {
+        throw new RuntimeException("EKS test without iam authenticator");
```
Comment (wyTrivail) [inline src/main/java/com/amazon/aocagent/helpers/EKSTestOptionsValidationHelper.java:62]: can we use baseexception?

File: src/main/java/com/amazon/aocagent/models/Context.java [hunk 5]
```diff
@@ -59,4 +59,18 @@
   /** Test resources region. */
   private String region;
 
+  /** EKS cluster attributes. */
+  private String eksEndpoint;
+
+  /** EKS cluster certificate. */
+  private String eksCertificate;
```
Comment (wyTrivail) [inline src/main/java/com/amazon/aocagent/models/Context.java:66]: what's this cert used for?

File: src/main/java/com/amazon/aocagent/testbeds/EKSTestBed.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+  @Override
+  public void init(Context context) {
+    EKSTestOptionsValidationHelper.checkEKSTestOptions(context);
+    this.context = context;
+  }
+
+  @Override
+  public Context launchTestBed() throws Exception {
+    return this.context;
```
Comment (wyTrivail) [inline src/main/java/com/amazon/aocagent/testbeds/EKSTestBed.java:19]: can we add a todo comment here to indicate we will make eks cluster creation in setup?

File: README.md (hunk 2 above)
Comment (mxiamxia) [reply README.md:80]: yes, can we add a little description for running this cmd? thx

File: src/main/java/com/amazon/aocagent/commands/CommonOption.java [hunk 7]
```diff
@@ -98,4 +93,23 @@ private Stack buildStack() throws IOException, BaseException {
     return mapper.readValue(
         new String(Files.readAllBytes(Paths.get(this.stackFilePath))), Stack.class);
   }
+
+  private void setExtraContext(Map<String, String> extra, Context context) {
```
Comment (wyTrivail) [inline src/main/java/com/amazon/aocagent/commands/CommonOption.java:97]: if i want to add a new test to eks, can i achieve it by just adding a mustache config, and send this config through command?

**Review by wyTrivail — CHANGES_REQUESTED** [review]

File: src/main/java/com/amazon/aocagent/testbeds/EKSTestBed.java (hunk 6 above)
Comment (hdj630) [reply src/main/java/com/amazon/aocagent/testbeds/EKSTestBed.java:19]: The current plan is to ask user to use their own cluster, they can simply use eksctl to create one.

File: src/main/java/com/amazon/aocagent/models/Context.java (hunk 5 above)
Comment (hdj630) [reply src/main/java/com/amazon/aocagent/models/Context.java:66]: https://docs.aws.amazon.com/eks/latest/APIReference/API_Certificate.html

File: src/main/java/com/amazon/aocagent/commands/CommonOption.java (hunk 7 above)
Comment (hdj630) [reply src/main/java/com/amazon/aocagent/commands/CommonOption.java:101]: OK, agree

**Pull request merged**